loki/batch/configure.py

```diff
@@ -0,0 +1,177 @@
+
+        if isinstance(routines, dict):
+            self.routines = CaseInsensitiveDict(routines)
+        else:
+            self.routines = CaseInsensitiveDict((r.name, r) for r in as_tuple(routines))
+
+        if isinstance(transformation_configs, dict):
+            self.transformation_configs = transformation_configs
+        else:
+            self.transformation_configs = dict((r.name, r) for r in as_tuple(transformation_configs))
+
+        # Resolve the dimensions for trafo configurations
+        for cfg in self.transformation_configs.values():
+            cfg.resolve_dimensions(dimensions)
+
+        # Instantiate Transformation objects
+        self.transformations = {
+            name: config.instantiate() for name, config in self.transformation_configs.items()
+        }
+
+    @classmethod
+    def from_dict(cls, config):
+        default = config['default']
+        routines = config.get('routines', [])
+        disable = default.get('disable', None)
+        enable_imports = default.get('enable_imports', False)
+
+        # Add any dimension definitions contained in the config dict
+        dimensions = config.get('dimensions', {})
+        dimensions = {k: Dimension(**d) for k, d in dimensions.items()}
+
+        # Create config objects for Transformation configurations
+        transformation_configs = config.get('transformations', {})
+        transformation_configs = {
+            name: TransformationConfig(name=name, **cfg)
+            for name, cfg in transformation_configs.items()
+        }
+
+        return cls(
+            default=default, routines=routines, disable=disable, dimensions=dimensions,
+            transformation_configs=transformation_configs, enable_imports=enable_imports
+        )
+
+    @classmethod
+    def from_file(cls, path):
+        import toml  # pylint: disable=import-outside-toplevel
+        # Load configuration file and process options
+        with Path(path).open('r') as f:
+            config = toml.load(f)
+
+        return cls.from_dict(config)
+
+
+class TransformationConfig:
+    """
+    Configuration object for :any:`Transformation` instances that can
+    be used to create :any:`Transformation` objects from dictionaries
+    or a config file.
+
+    Parameters
+    ----------
+    name : str
+        Name of the transformation object
+    module : str
+        Python module from which to load the transformation class
+    classname : str, optional
+        Name of the class to look for when instantiating the transformation.
+        If not provided, ``name`` will be used instead.
+    path : str or Path, optional
+        Path to add to the sys.path before attempting to load the ``module``
+    options : dict
+        Dicts of options that define the transformation behaviour.
+        These options will be passed as constructor arguments using
+        keyword-argument notation.
+    """
+
+    _re_dimension = re.compile(r'\%dimensions?\.(.*)\%')
```

Comment:
Two questions about the regex:
- is the `s` of `dimensions` really intended to be optional?
- I think the capture group should be lazy, otherwise something like `%dimensions.vertical% + %dimensions.horizontal%` would be matched as one rather than two occurences (for whichever reason one might write that)

```suggestion
    _re_dimension = re.compile(r'\%dimensions\.(.*?)\%')
```

Reply:
Hehe, the optional `s` was lazyness (I kept getting it wrong...). Removed now, as we should aim for consistency. :smirk: